bfdd/bfd_packet.c

```diff
@@ -1257,3 +1567,37 @@ int bp_echov6_socket(const struct vrf *vrf)
 
 	return s;
 }
+
+/* get peer's mac address to be used with Echo packets */
+void bfd_peer_mac_set(int sd, struct bfd_session *bfd,
+		      struct sockaddr_any *peer, struct interface *ifp)
+{
+	struct arpreq arpreq_;
+
+	if (CHECK_FLAG(bfd->flags, BFD_SESS_FLAG_MAC_SET))
+		return;
+
+	/* if BFD session is not up get Peer's Mac Address */
+	if (peer->sa_sin.sin_family == AF_INET) {
+		/* IPV4 */
+		struct sockaddr_in *addr =
+			(struct sockaddr_in *)&arpreq_.arp_pa;
+
+		bzero(&arpreq_, sizeof(struct arpreq));
+		addr->sin_family = AF_INET;
+		memcpy(&addr->sin_addr.s_addr, &peer->sa_sin.sin_addr, 4);
+		strcpy(arpreq_.arp_dev, ifp->name);
+
+		if (ioctl(sd, SIOCGARP, &arpreq_) < 0) {
+			zlog_warn("BFD: getting peer's mac failed error %s",
+				  strerror(errno));
+			UNSET_FLAG(bfd->flags, BFD_SESS_FLAG_MAC_SET);
+			memset(bfd->peer_hw_addr, 0, ETH_ALEN);
+
+		} else {
+			memcpy(bfd->peer_hw_addr, arpreq_.arp_ha.sa_data,
```

Comment:
Please use `sizeof(bfd->peer_hw_addr)` instead of `ETH_ALEN`.

Reply:
done